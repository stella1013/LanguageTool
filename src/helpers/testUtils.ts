//import { ConnectedRouter } from 'connected-react-router';
//import React from 'react';
//import { Provider } from 'react-redux';
////import { mergeDeepRight } from 'ramda';
//import rootReducer from '../redux/reducers';
////import  createStoreWithMiddleWare  from '../redux/store';
//import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
//import { AppState } from '../redux/store';
//import ReduxThunk  from 'redux-thunk';
//import { createStore, applyMiddleware } from 'redux';
//import rootReducer from '../redux/reducers';
//import thunk from 'redux-thunk';

/*
	* Returns Shallow Wrapper containing node(s) with the given data-test value
	* @param {Shallow Wrapper} - Enzyme shallow wrapper to search within
	* @param {string} val - Value of data-test attribute for search.
	* @returns {Shallow Wrapper}
	*/ 
	export const findByTestAttr = (wrapper:any, val:string) => {
		return wrapper.find(`[data-test="${val}"]`);
	}

	export const storeFactory = (initialState:any) =>{
		const createMockStore = configureMockStore();
		return createMockStore(initialState);
	}
	/*
	export const makeMountRender = (Component:any, defaultProps = {}) => {
		return (customProps = {}) => {
		  const props = {
			...defaultProps,
			...customProps
		  };
		  return mount(<Component {...props} />);
		};
	  };

	  export const makeStore = (customState = {}) => {
		const root = rootReducer({}, { type: '@@INIT' });
		const state = mergeDeepRight(root, customState);
		return createStoreWithMiddleWare(rootReducer, state);
	  };
	  export const snapshotify = (reactWrapper:any) => {
		return reactWrapper.html();
	  };

	  export const reduxify = (Component, props = {}, state = {}) => {
		return function reduxWrap() {
		  return (
			<Provider store={makeStore(state)}>
			  <ConnectedRouter history={history}>
				<Component {...props} />
			  </ConnectedRouter>
			</Provider>
		  );
		}
	  };

*/