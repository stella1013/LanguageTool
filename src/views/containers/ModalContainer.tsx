import * as React from 'react';
import { connect } from 'react-redux';
import { AppState } from '../../redux/store';
import { modalOperations } from '../../redux/Modal';
import { Modal, Button } from 'react-bootstrap';


class ModalContainer extends React.Component<ReduxType> {
	modalHandler = ()=>{
        if(!this.props.isModalOpen){
            this.props.showModal(true);
        }else{
            this.props.showModal(false);
        }
    }
	render() {
		return (
			<div data-test="component-modal">
				<Modal show={this.props.isModalOpen} onHide={():void=>this.modalHandler()}>
					<Modal.Header closeButton>
						<Modal.Title>General File Upload Guidelines</Modal.Title>
					</Modal.Header>

					<Modal.Body>
						<h5>The specifications for file uploads are:</h5>
						<ul>
							<li>up to 10MB in size</li>
							<li>
								the file name can only have the following characters: a-z A-Z
								0-9 _ . -
							</li>
							<li> the file name can be no longer than 512 characters</li>
							<li>the file name can only have a single '.'</li>
						</ul>
						<h5>Image Specifications:</h5>
						<ul>
							<li>jpeg/jpg or png formats allowed</li>
							<li>Maximum image dimensions 2048px width and 2048px height</li>
							<li>72dpi</li>
						</ul>
						<h5>Video Specifications:</h5>
						<ul>
							<li>mpeg/mpg format allowed</li>
							<li>up to 50MB in size</li>
							<li>Maximum image dimensions 2048px width and 2048px height</li>
							<li>72dpi</li>
						</ul>
					</Modal.Body>

					<Modal.Footer>
						<Button variant="secondary" onClick={this.modalHandler}>
							Close
						</Button>
					</Modal.Footer>
				</Modal>
			</div>
		);
	}
}

const mapStateToProps = (state:AppState) => {
	return {
		title:state.appTitle.appTitle,
		isModalOpen:state.modal.showModal
	};
};
const mapDispatchToProps = (
	dispatch: any
) => {
	return {
		showModal: (showModal:boolean) => dispatch(modalOperations.showModal(showModal))
	};
};
type ReduxType = ReturnType<typeof mapStateToProps> &
	ReturnType<typeof mapDispatchToProps>;

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(ModalContainer);
