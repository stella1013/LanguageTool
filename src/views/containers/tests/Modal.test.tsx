import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import configureMockStore from 'redux-mock-store';
import { findByTestAttr } from '../../../helpers/testUtils';
import thunk from 'redux-thunk';
import ModalContainer from '../ModalContainer';
import { Provider } from 'react-redux';



describe('Modal', () => {
	/*
	 * Factory function to create a Shallow Wrapper for the App Component
	 * @function setup
	 * @param {object} props - component props
	 * @param {object} state - initial state
	 * @returns {Shallow Wrapper}
	 */
	const createMockStore = configureMockStore([thunk]);
	const initialState = {};
	//const store = createMockStore(initialState);
	let wrapper: any;
	describe('rendering', () => {
		let props ={
			appTitle:'title'
		}
		beforeEach(() => {
			const store = createMockStore(initialState);
			wrapper = shallow(<Provider store={store}><ModalContainer  /></Provider>)
			
		});

		it('renders without crashing', () => {
			const component = findByTestAttr(wrapper, 'component-modal');
			expect(component).toBeTruthy();
		});

		it('render matches snapshot', () => {
			expect(wrapper).toMatchSnapshot();
		});

		
	});

	describe('state updates', () => {});
});
