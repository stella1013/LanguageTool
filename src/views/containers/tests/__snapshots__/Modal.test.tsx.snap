// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal rendering render matches snapshot 1`] = `ShallowWrapper {}`;
